bazel/test.bzl

```diff
@@ -130,6 +130,53 @@ def _redpanda_cc_test(
         local_defines = local_defines,
     )
 
+def _redpanda_cc_fuzz_test(
+        name,
+        timeout,
+        srcs = [],
+        defines = [],
+        deps = [],
+        custom_args = [],
+        env = {},
+        data = []):
+    """
+    Helper to define a Redpanda C++ fuzzing test.
+
+    Args:
+      name: name of the test
+      timeout: same as native cc_test
+      srcs: test source files
+      defines: definitions of object-like macros
+      deps: test dependencies
+      custom_args: arguments from cc_test users
+      env: environment variables
+      data: data file dependencies
+    """
+    native.cc_test(
+        name = name,
+        timeout = timeout,
+        srcs = srcs,
+        defines = defines,
+        deps = deps,
+        copts = redpanda_copts(),
+        args = custom_args,
+        features = [
+            "layering_check",
+        ],
+        tags = [
+            "fuzz",
+        ],
+        env = env,
+        data = data,
+        cxxopts = [
+            "-g",
```

Comment:
@oleiman debug info is explicitly being turned on for the fuzz target

Reply:
oh nice. missed that 👍

Reply:
Do we need a `--config` for this so that `-g` is applied globally instead of only to this target? We need all libraries we link against to be `-g` too right?

Reply:
oh... yes. Of course we do :sweat_smile: